netket/graph/lattice.py

```diff
@@ -20,8 +20,15 @@
 import networkx as _nx
 import warnings
 
+cutoff_tol = 1e-5
+"""Tolerance for the maximum distance cutoff when computing the sparse distance matrix.
+This is necessary because of floating-point errors when computing the distance in non-trivial 
+lattices.
+"""
+
 
 def get_edges(atoms_positions, cutoff):
+    cutoff = cutoff + cutoff_tol
```

Comment:
This innocent-looking change changes the time it takes for `Test/Graph/test_graph.py::test_automorphisms` by _a lot_:
```
# before:
2.13s call     Test/Graph/test_graph.py::test_automorphisms
# after:
226.23s call     Test/Graph/test_graph.py::test_automorphisms
```

This is probably because the lattice graphs previously did not recognize all edges. In any case, we should probably fix this test.

By the way, is there a reason `test_automorphism` is only run on the lattices? Also, do we really need this test in the current form? At the moment, it only seems to check that two different `networkx` methods return the same results (as all our graphs use networkx anyway for this).

Reply:
I'm all in for writing better tests.
I don't know who wrote them originally.

Reply:
There is overlap with what is being discussed [here](https://github.com/netket/netket/pull/633#discussion_r616672597)

Reply:
I think for now we could honestly just remove that test. At least to some extent the `NetworkX.autmorphisms` method at least is covered in the symmetry group tests as well, so we would not lose significant coverage.

Reply:
can you open a PR for this?